clang/lib/3C/CastPlacement.cpp

```diff
@@ -266,6 +270,23 @@ void CastPlacementVisitor::reportCastInsertionFailure(
   ErrorBuilder.AddString(CastStr);
 }
 
+void CastPlacementVisitor::updateRewriteStats(CastNeeded CastKind) {
+  auto &PStats = Info.getPerfStats();
+  switch (CastKind) {
+  case CAST_NT_ARRAY:
+    PStats.incrementNumCheckedCasts();
+    break;
+  case CAST_TO_WILD:
+    PStats.incrementNumWildCasts();
+    break;
+  case CAST_TO_CHECKED:
+    PStats.incrementNumAssumeBounds();
+    break;
```

Comment:
I don't understand the different cast kinds. If you are sure these are correct, ...

Reply:
I computed these from `CastPlacementVisitor.cpp`. I am fairly sure they are okay, but @john-h-kastner could quickly check this.